File: pset2/caesar/caesar.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
// Get cipher code from command line, print user input (plaintext) in cipher form (ciphertext)
int main(int argc, string argv[])
{
    if (argc == 2)
    {
        // convert argv[1] to integer
        int k = atoi(argv[1]);

        // get string from user input
        string plaintext = get_string("plaintext: ");

        printf("ciphertext: ");
        // encipher string by printing
        for (int c = 0; c < strlen(plaintext); c++)
        {
            // handle uppercase chars
            if (isalpha(plaintext[c]) && isupper(plaintext[c]))
            {
                printf("%c", ((plaintext[c] - 65 + k) % 26) + 65);
            }
            // handle lowercase chars
            else if (isalpha(plaintext[c]) && islower(plaintext[c]))
            {
                printf("%c", ((plaintext[c] - 97 + k) % 26) + 97);
            }
            // handle all other chars
            else
            {
                printf("%c", plaintext[c]);
            }
        }
        printf("\n");
        exit(0);
    }
    else
    {
        printf("Usage: %s [number]\n", argv[0]);
        exit(1);
    }
}

File: pset6/sentimental/caesar/caesar.py
# filename: caesar.py
"""
Get cipher code from command line,
print user input (plaintext) in cipher form (ciphertext).
"""
import sys

# check whether command line argument is given
if sys.argv[1].isdecimal():
    # convert command line string to int
    # and use modulo in case the number is larger than 26
    k = int(sys.argv[1]) % 26

    # prompt user for plaintext
    plaintext = str(input("plaintext: "))

    # initialize empty string variable to contain ciphertext
    ciphertext = ''

    # encipher plaintext.
    for c in plaintext:
        # add enciphered alphabetical characters to ciphertext variable
        if c.isalpha():
            if c.isupper():
                ciphertext += chr(((ord(c) - 65 + k) % 26) + 65)
            elif c.islower():
                ciphertext += chr(((ord(c) - 97 + k) % 26) + 97)
        else:
            # add non-alphabetical characters without enciphering
            ciphertext += c
    # display ciphertext
    print('ciphertext:', ciphertext)
else:
    # exit and display program usage to user
    print("Usage:", sys.argv[0], 'k')